### PR TITLE
fix(Build): Fix "empty.c file not found" Error on Make v3.81 or Older

### PR DESCRIPTION
**Pull request by Jake-Carter**
### Description

The [`$(file)` operation](https://www.gnu.org/software/make/manual/make.html#File-Function) that was being used to generate our `project_defines.h` header file (as of #628) was only supported on Make v3.82 or newer.  As a result, v3.81 or older would throw the following error:

```bash
arm-none-eabi-gcc.exe: error: C:/Users/JCarter3/repos/fork/msdk/Examples/MAX78000/CNN/mnist-riscv/build/_empty_tmp_file.c: No such file or directory
arm-none-eabi-gcc.exe: fatal error: no input files
compilation terminated.
make[1]: *** [/C/Users/JCarter3/repos/fork/msdk/Libraries/CMSIS/Device/Maxim/GCC/gcc.mk:508: /c/Users/JCarter3/repos/fork/msdk/Examples/MAX78000/CNN/mnist-riscv/build/project_defines.h] Error 1
make: *** [Makefile:56: arm] Error 2
```

This PR replaces the operation with an `echo` command that works cross-platform.

### Checklist Before Requesting Review

- [x] PR Title follows correct guidelines.
- [x] Description of changes and all other relevant information.
- [x] (Optional) Link any related Github issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

File: Libraries/CMSIS/Device/Maxim/GCC/gcc.mk [hunk 1]
```diff
@@ -503,8 +503,8 @@ debug:
...
 $(BUILD_DIR)/project_defines.h: mkbuildir
-	$(file > $(BUILD_DIR)/empty.c,)
-	$(file > $(BUILD_DIR)/project_defines.h,// This is a generated file that's used to detect definitions that have been set by the compiler and build system.)
-	@$(CC) -E -P -dD $(BUILD_DIR)/empty.c $(CFLAGS) >> $(BUILD_DIR)/project_defines.h
-	@rm $(BUILD_DIR)/empty.c
-	@rm empty.d
\ No newline at end of file
+	@echo "" > $(BUILD_DIR)/_empty_tmp_file.c
+	@echo "// This is a generated file that's used to detect definitions that have been set by the compiler and build system." > $@
+	@$(CC) -E -P -dD $(BUILD_DIR)/_empty_tmp_file.c $(CFLAGS) >> $(BUILD_DIR)/project_defines.h
```
Comment (rotx-maxim) [inline Libraries/CMSIS/Device/Maxim/GCC/gcc.mk:508]: Should we change $(BUILD_DIR)/project_defines.h to $@ for consistency?

**Comment by sihyung-maxim** [comment]
Do we want to add a "Build" scope for cases like these in the PR Title Checker?

**Comment by Jake-Carter** [comment]
> Do we want to add a "Build" scope for cases like these in the PR Title Checker?

Tried to in https://github.com/Analog-Devices-MSDK/msdk/pull/657/commits/73037d768567ef678530a465f20c0e529b157099, do I need to do something else?

**Comment by khpeterson** [comment]
I was just preparing this very same pull request - gcc_riscv.mk has the same problem.

**Comment by Jake-Carter** [comment]
> I was just preparing this very same pull request - gcc_riscv.mk has the same problem.

Thanks @khpeterson, applied to the RISCV file in https://github.com/Analog-Devices-MSDK/msdk/pull/657/commits/c6f5240536991e38f62c3804b9aecf1384462293

**Comment by sihyung-maxim** [comment]
> > Do we want to add a "Build" scope for cases like these in the PR Title Checker?
> 
> Tried to in [73037d7](https://github.com/Analog-Devices-MSDK/msdk/commit/73037d768567ef678530a465f20c0e529b157099), do I need to do something else?

Nope, that should be it for the workflow. I think because the runner references the workflow file that's in the main branch, the action is going to fail and the new **Build** scope won't be visible until this PR is merged.

**Review by sihyung-maxim — APPROVED** [review]

**Review by Jacob-Scheiffler — APPROVED** [review]

**Comment by Jake-Carter** [comment]
Added regression test for building with make v3.81 (https://github.com/Analog-Devices-MSDK/msdk/pull/657/commits/ecf103b3fe424490be87260158e78026545b2943)

**Pull request merged**